include/nicegraf.h

```diff
@@ -111,128 +111,6 @@ extern "C" {
 #pragma region ngf_type_declarations
 #endif
 
-/**
```

Comment:
`ngf_device_capabilities` uses `ngf_sample_count` as a variable, but its declared later. Need to move `ngf_device_capabilities` to the bottom of the structs definitions, and this also means we need to move `ngf_device` itself because it references `ngf_device_capabilities`